src/moduls/render.ts

```diff
@@ -0,0 +1,2146 @@
+
+    renderCards(flowers: Icard[]) {
+        const newdiv = document.createElement('div');
+        newdiv.classList.add('ggd');
+        cardsD.length = 0;
+        flowers.forEach(function (key) {
+            const cards = `<div class="catalog-card">
+		<div class="card-item">
+			<div class="img-item">
+				<img class="img-buket" src="${key.img}" alt="image">
+			</div>
+			<div class="text-item">
+			${key.name}
+			</div>
+			<div class="much">
+				Количество:${key.much}
+			</div>
+			<div class="size">
+			Бутон ${key.but}
+			</div>
+			<div class="fcolor">
+				Цвет: ${key.color}
+			</div>
+			<div class="type">
+				Тип: ${key.type}
+			</div>
+			<div class="length">
+				Длина цветка: ${key.flength}
+			</div>
+			<div class="length">
+				Популярный: ${key.popular}
+			</div>
+		</div>
+	</div>`;
+
+            newdiv.insertAdjacentHTML('afterbegin', cards);
+
+            cardsD.push(key);
+
+            return newdiv;
+        });
+
+        card.replaceChildren(newdiv);
+
+        const cart = document.querySelector('.items') as HTMLTemplateElement;
+        let a = 0;
+        for (const item of Object(document.getElementsByClassName('catalog-card'))) {
+            if (item.style.display !== 'none')
+                item.addEventListener('click', function () {
+                    if (item.classList.contains('active')) {
+                        item.classList.remove('active');
+                        a--;
+                        cart.innerHTML = `${a}`;
+                    } else {
+                        item.classList.add('active');
+                        a++;
+                        cart.innerHTML = `${a}`;
+                    }
+                    if (a > 10) {
+                        alert('Извините, все слоты заполнены');
+                        item.classList.remove('active');
+                        a = 10;
+                    }
+                });
+        }
+    }
+
+    renderCardsD(flowers: Icard[]) {
+        const newdiv = document.createElement('div');
+        newdiv.classList.add('ggd');
+
+        flowers.forEach(function (key) {
+            const cards = `<div class="catalog-card">
+		<div class="card-item">
+			<div class="img-item">
+				<img class="img-buket" src="${key.img}" alt="image">
+			</div>
+			<div class="text-item">
+			${key.name}
+			</div>
+			<div class="much">
+				Количество:${key.much}
+			</div>
+			<div class="size">
+			Бутон ${key.but}
+			</div>
+			<div class="fcolor">
+				Цвет: ${key.color}
+			</div>
+			<div class="type">
+				Тип: ${key.type}
+			</div>
+			<div class="length">
+				Длина цветка: ${key.flength}
+			</div>
+			<div class="length">
+				Популярный: ${key.popular}
+			</div>
+		</div>
+	</div>`;
+
+            newdiv.insertAdjacentHTML('afterbegin', cards);
+
+            return newdiv;
+        });
+
+        card.replaceChildren(newdiv);
+
+        const cart = document.querySelector('.items') as HTMLTemplateElement;
+        let a = 0;
```

Comment:
Using this variable leads to the incorrect behavior for the number in cart. The number in the cart will become 0 every time this function is called. 
- every time we apply a fiter or a sorting, the renderCardsD function will be called;
- the function will create new cards and will add "onclick" event handlers for them;
- in every event handler there will be a closure to this variable, which will be newly created every time we run the renderCardsD function